controllers/fluentbitconfig_controller.go

```diff
@@ -59,112 +63,239 @@ type FluentBitConfigReconciler struct {
+			if err != nil {
+				return ctrl.Result{}, err
 			}
-			for _, s := range scripts {
-				sec.Data[s.Name] = []byte(s.Content)
+
+			// Create or update the corresponding Secret
+			sec := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cfg.Name,
+					Namespace: ns,
+				},
 			}
-			sec.SetOwnerReferences(nil)
+
 			if err := ctrl.SetControllerReference(&cfg, sec, r.Scheme); err != nil {
-				return err
+				return ctrl.Result{}, err
 			}
-			return nil
-		}); err != nil {
-			return ctrl.Result{}, err
-		}
 
-		r.Log.Info("Fluent Bit main configuration has updated", "logging-control-plane", ns, "fluentbitconfig", cfg.Name, "secret", sec.Name)
+			if _, err := controllerutil.CreateOrPatch(ctx, r.Client, sec, func() error {
+				sec.Data = map[string][]byte{
+					"fluent-bit.conf": []byte(mainAppCfg),
+					"parsers.conf":    []byte(parserCfg),
+				}
+				for _, s := range scripts {
+					sec.Data[s.Name] = []byte(s.Content)
+				}
+				sec.SetOwnerReferences(nil)
+				if err := ctrl.SetControllerReference(&cfg, sec, r.Scheme); err != nil {
+					return err
+				}
+				return nil
+			}); err != nil {
+				return ctrl.Result{}, err
+			}
+
+			r.Log.Info("Fluent Bit main configuration has updated", "logging-control-plane", ns, "fluentbitconfig", cfg.Name, "secret", sec.Name)
+		}
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *FluentBitConfigReconciler) processNamespaceFluentBitCfgs(ctx context.Context, fb fluentbitv1alpha2.FluentBit, inputs fluentbitv1alpha2.ClusterInputList) ([]fluentbitv1alpha2.FilterList, []fluentbitv1alpha2.OutputList,
+	[]fluentbitv1alpha2.ParserList, []fluentbitv1alpha2.ClusterParserList, []string, error) {
+	var nsCfgs fluentbitv1alpha2.FluentBitConfigList
+	var filters []fluentbitv1alpha2.FilterList
+	var outputs []fluentbitv1alpha2.OutputList
+	var parsers []fluentbitv1alpha2.ParserList
+	var clusterParsers []fluentbitv1alpha2.ClusterParserList
+	var rewriteTagConfigs []string
+	// set of rewrite_tag plugin configs to mutate tags for log records coming out of a namespace
+	selector, err := metav1.LabelSelectorAsSelector(&fb.Spec.NamespaceFluentBitCfgSelector)
+	if err != nil {
+		return filters, outputs, parsers, clusterParsers, nil, err
+	}
+
+	if err := r.List(ctx, &nsCfgs, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		return filters, outputs, parsers, clusterParsers, nil, err
+	}
+
+	// Form a slice of list of resources per namespace
+	for _, cfg := range nsCfgs.Items {
+		filterList, outputList, parserList, clusterParserList, err := r.ListNamespacedResources(ctx, cfg)
+		if err != nil {
+			return filters, outputs, parsers, clusterParsers, nil, err
+		}
+		filters = append(filters, filterList)
+		outputs = append(outputs, outputList)
+		parsers = append(parsers, parserList)
+		clusterParsers = append(clusterParsers, clusterParserList)
+
+		rewriteTagConfig := r.generateRewriteTagConfig(cfg, inputs)
+		if rewriteTagConfig != "" {
```

Comment:
What if we add other input components in the follow-up? 
https://docs.fluentbit.io/manual/pipeline/inputs
We plan to add more input components in a later version.

Reply:
See https://docs.fluentbit.io/manual/pipeline/filters/kubernetes#workflow-of-tail-+-kubernetes-filter
The input component mainly responsible for collection of container/pod/namespace level logs is Tail, so that's why I'm checking for it and fetching the tag from it.

Moving forward if there are other inputs introduced in FluentBit itself that can collect namespace level logs we can make the logic generic for multiple inputs.

Reply:
Also this is what Fluent-Operator has https://github.com/fluent/fluent-operator/blob/master/charts/fluent-operator/templates/fluentbit-clusterfilter-kubernetes.yaml

This filter config maps to Tail input config.